### PR TITLE
refactor: Many more small refactors

### PR DESCRIPTION
**Pull request by deluca-mike**
- library errors moved to interfaces so they can be inherited
- payload sizes now typed as `uint32`
- `_toAddress` replaced with non-assembly code via a `RegistryParameters` library
- `PayloadBroacaster` no longer updates parameters upon initialization, in order to simplify and decouple the deployment process
- `RegistryParameters` used everywhere to fetch, check bounds, and safely cast parameters as needed
- imports sorted
- `IParameterRegistryLike` no longer needed by any contracts other than the `RegistryParameters` library
- some corrected natspec
- adding/removing nodes to/from canonical network is now idempotent (i.e. will not revert if no change)
- max canonical nodes is now a parameter from the parameter registry
- `depositWithPermit` added to `PayerRegistry`
- pagination when fetching rates from the `RateRegistry` is now caller-controlled (i.e. caller specifies the count)
- `AppchainToken` renamed to `FeeToken`
- `NodeRegistry` no longer updates `maxCanonicalNodes` upon initialization, in order to simplify and decouple the deployment process
- `NodeRegistryStorage` reorganized for better packing
- `PayerRegistry` no longer updates parameters upon initialization, in order to simplify and decouple the deployment process
- deployment tests updated and simplified given less coupling of contracts to parameter registries at initialization
- all mock contracts deleted in favour of full `vm.mock` usage
- test coverage added for parameters being out of bounds for expected usage

## Summary by CodeRabbit

- **New Features**
  - Introduced a `RegistryParameters` library for safer, type-checked parameter access and updates across contracts.
  - Added support for deposits using ERC20 permit signatures.
  - Added new functions for updating and retrieving parameters in several registries and broadcasters.

- **Refactor**
  - Centralized parameter fetching and updating through the new `RegistryParameters` library, simplifying contract logic and improving type safety.
  - Updated storage types for payload sizes and other parameters to more efficient types (e.g., `uint32`).
  - Renamed and repurposed the "Appchain Token" to "Fee Token" with updated interface, storage, and naming.
  - Standardized interface inheritance to include new error interfaces for better error handling.
  - Replaced direct parameter registry calls and manual conversions with library-based, type-checked access.
  - Simplified deployment salt validations by using integer zero checks instead of zeroed bytes32 values.
  - Removed internal helper functions for parameter conversions and replaced with library calls.
  - Adjusted contract initialization and update flows to rely on parameter registry values.

- **Bug Fixes**
  - Improved bounds checking for parameter values, reverting on out-of-type-bounds values.
  - Updated validation checks for zero values in deployment and proxy logic.
  - Modified certain revert conditions to return silently for idempotent operations.

- **Tests**
  - Expanded and refactored test suites to cover new parameter bounds, migration edge cases, and permit-based deposits.
  - Added comprehensive tests for the new `RegistryParameters` library.
  - Updated all tests to use new mocking utilities and type-checked parameter retrieval.
  - Replaced mock contract deployments with fixed deterministic addresses for improved test determinism.
  - Adjusted tests to reflect renamed contracts, interfaces, and updated parameter management.
  - Improved migration tests to simulate revert data and parameter bounds errors.

- **Chores**
  - Removed unused mock contracts and redundant test code.
  - Reorganized imports and cleaned up code for clarity and maintainability.
  - Standardized error handling by importing error interfaces and replacing local error declarations.
  - Updated encoding in mocks to use signature-based calls for consistency.

- **Documentation**
  - Updated interface and function comments to reflect new parameter management and error handling approaches.

### CODE REVIEW

File: test/unit/Factory.t.sol [hunk 1]
```diff
@@ -142,8 +142,8 @@ contract FactoryTests is Test {
     /* ============ computeProxyAddress ============ */
 
     function test_computeProxyAddress() external view {
-        address expectedProxy_ = _getExpectedProxy(_alice, bytes32(0));
-        assertEq(_factory.computeProxyAddress(_alice, bytes32(0)), expectedProxy_);
+        address expectedProxy_ = _getExpectedProxy(_alice, 0);
+        assertEq(_factory.computeProxyAddress(_alice, 0), expectedProxy_);
     }
```
[the remark this answers is not in the document]
Comment (deluca-mike) [reply test/unit/Factory.t.sol:147]: Salts are better as `bytes32` in this codebase, and also the factory's `computeProxyAddress` function is `computeProxyAddress(address,bytes32)`.

File: src/abstract/PayloadBroadcaster.sol [hunk 2]
```diff
@@ -71,34 +72,47 @@
...
 
     /// @inheritdoc IPayloadBroadcaster
-    function initialize() public virtual initializer {
-        // Since both the min and max start at 0, the max must be updated before the min, since the min can never be
-        // set to be greater than the max.
-        _updateMaxPayloadSize();
-        _updateMinPayloadSize();
-        _updatePauseStatus(); // The contract may start out paused, as needed.
-    }
+    function initialize() public virtual initializer {}
```
[the remark this answers is not in the document]
Comment (deluca-mike) [reply src/abstract/PayloadBroadcaster.sol:76]: This is fine. Once the rest of the protocol is deployed (or even if it already is), the various update functions can be called immediately so that the contract can operate as intended.

File: src/abstract/PayloadBroadcaster.sol [hunk 3]
```diff
@@ -71,34 +72,47 @@
...
-        if (!_updateMinPayloadSize()) revert NoChange();
+        uint32 minPayloadSize_ = RegistryParameters.getUint32Parameter(parameterRegistry, minPayloadSizeParameterKey());
+
+        PayloadBroadcasterStorage storage $ = _getPayloadBroadcasterStorage();
+
+        if (minPayloadSize_ > $.maxPayloadSize) revert InvalidMinPayloadSize();
+        if (minPayloadSize_ == $.minPayloadSize) revert NoChange();
+
+        emit MinPayloadSizeUpdated($.minPayloadSize = minPayloadSize_);
     }
```
[the remark this answers is not in the document]
Comment (deluca-mike) [reply src/abstract/PayloadBroadcaster.sol:89]: It is not clear if an empty payload is disallowed at the application layer. I will leave this comment unresolved and wait for the team's input.

File: src/settlement-chain/PayerRegistry.sol [hunk 4]
```diff
@@ -96,25 +96,44 @@
...
+                amount_,
+                deadline_,
+                v_,
+                r_,
+                s_
+            )
+        );
+        // slither-disable-end unchecked-lowlevel
+        // slither-disable-end low-level-calls
+
```
[the remark this answers is not in the document]
Comment (deluca-mike) [reply src/settlement-chain/PayerRegistry.sol]: This is fine. We need to account for the possibility that the permit was already used out-of-band, and just try to use it, and carry on. If we require the permit be successful here, then a griefer can front-run this transaction with a call to the token contract with just the permit, and thus cause the user's depoit transaction to fail.

**Review by fbac — APPROVED** [review]

File: src/abstract/PayloadBroadcaster.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (deluca-mike) [reply src/abstract/PayloadBroadcaster.sol:82]: Can you document this as an issue so the team can revisit it in the future?
